externs/shaka/ads.js

```diff
@@ -37,73 +37,95 @@ shaka.extern.IAdManager.Factory;
 shaka.extern.IAd = class {
   /**
    * @return {number}
+   * @export
```

Comment:
I don't understand why you've added export annotations to the externs here.  They will not do anything in the Closure compiler.

Reply:
They won't in the closure compiler, but as the shaka-player typings themself refer to this type, it has to be defined somewhere. By "exporting" the externs, the ts-generator can include this type in the generated typings, therefore making sure the TypeScript compiler has access to a definition of this type when checking the shaka-player types.

Reply:
I'd have preferred to add a custom annotation, e.g. `@tsexport`, but Closure errors when encountering unknown annotations.

Reply:
Well, I've finally discovered the `--extra_annotation_name` parameter for closure. I will replace these exports with a new `@exportTypescript` annotation.